libs/shared/form-fields/src/lib/InputController/InputController.tsx

```diff
@@ -25,10 +29,68 @@ export const InputController: FC<Props> = ({
   name = id,
   placeholder,
   textarea,
+  currency,
   type = 'text',
+  format,
   onChange,
 }) => {
   const { register } = useFormContext()
+
+  if (currency) {
+    return (
+      <Controller
+        name={name}
+        render={({ value, onChange }) => (
+          <NumberFormat
+            customInput={Input}
+            id={id}
+            disabled={disabled}
+            name={name}
+            placeholder={placeholder}
+            label={label}
+            type="text"
+            decimalSeparator=","
+            thousandSeparator="."
+            suffix=" kr."
+            value={value}
+            format={format}
+            onValueChange={({ value }) => {
+              onChange(value)
+            }}
+            hasError={error !== undefined}
+            errorMessage={error}
+          />
+        )}
+      />
+    )
+  }
+
+  if (format && ['text', 'tel'].includes(type)) {
```

Comment:
hmm shouldn't `number` also be in the array? Like bank info is essentially of type number and has a format `####-##-######`